apps/src/gamelab/GameLabSprite.js

```diff
@@ -322,9 +322,9 @@ var _collideWith = function (p5Inst, type, target, callback) {
   if (state.__i < state.__others.length) {
     state.__result = this._collideWithOne(type, state.__others[state.__i], callback) || state.__result;
     // Not done, unless we're on the last item in __others:
-    state.doneExec = state.__i >= (state.__others.length - 1);
+    state.doneExec_ = state.__i >= (state.__others.length - 1);
   } else {
-    state.doneExec = true;
+    state.doneExec_ = true;
   }
   return state.__result;
```

Comment:
`_collideWith` is a native function that invokes an interpreter callback, so it's rigged to be suspendable/resumable in case the interpreter hits its step limit inside the callback.

`_createGroupState` is a native function that invokes `_collideWith` in a loop, so it in turn can also be suspended - it needs to track its own state, and the state of `_collideWith` in a way that won't overalp, to be able to pick up where it left off.

It would be much better if this was all done in a generic way.  [Here's some old historical context](https://github.com/code-dot-org/code-dot-org/pull/6880).  I'm not sure what that generic approach is, but I suspect we should merge this fix and then sit down together and work out a proper solution to this.  @joshlory and I have spent time on this before and mostly ended up with the expensive solution of running the interpreter in a webworker or other context where we can effectively 'block' in native code, but that doesn't enable debugging properly either.  We can also push this code into the interpreter, but that would be impractically slow for collision detection in gamelab.

In this situation it's fairly common for the interpreter to hit its step limit inside the callback

See the comments on [`_collideWith`](https://github.com/code-dot-org/code-dot-org/blob/78280515a6cfa0e9f3ca4d6c2ef0bc096f2617f6/apps/src/gamelab/GameLabSprite.js#L260-L278) and [`_groupCollideGameLab`](https://github.com/code-dot-org/code-dot-org/blob/e65be98060360cee6c7d93b173d3ad8ef01dc0c2/apps/src/gamelab/GameLabGroup.js#L10-L36).